dotcom-rendering/src/footballMatches.ts

```diff
@@ -33,14 +45,355 @@ export type FootballMatch = MatchResult | MatchFixture | LiveMatch;
+	}
+
+	return ok({
+		kind: 'Fixture',
+		homeTeam: feFixture.homeTeam.name,
+		awayTeam: feFixture.awayTeam.name,
+		dateTime: date.value,
+		paId: feFixture.id,
+	});
+};
+
+const parseMatchResult = (
+	feResult: FEResult | FEMatchDay,
+): Result<ParserError, MatchResult> => {
+	if (feResult.type === 'MatchDay' && !feResult.result) {
+		return error({
+			kind: 'ExpectedMatchDayResult',
+			message: `A result match day must have 'result' set to 'true', but this was not the case for ${feResult.id}`,
+		});
+	}
+
+	const date = parseMatchDate(feResult.date);
+
+	if (date.kind === 'error') {
+		return error({ kind: 'FootballMatchInvalidDate', message: date.error });
+	}
+
+	const homeScore = parseScore(feResult.homeTeam, 'Result');
+
+	if (homeScore.kind === 'error') {
+		return homeScore;
+	}
+
+	const awayScore = parseScore(feResult.awayTeam, 'Result');
+
+	if (awayScore.kind === 'error') {
+		return awayScore;
+	}
+
+	return ok({
+		kind: 'Result',
+		homeTeam: {
+			name: feResult.homeTeam.name,
+			score: homeScore.value,
+		},
+		awayTeam: {
+			name: feResult.awayTeam.name,
+			score: awayScore.value,
+		},
+		dateTime: date.value,
+		paId: feResult.id,
+		comment: feResult.comments,
+	});
+};
+
+const parseLiveMatch = (
+	feMatchDay: FEMatchDay,
+): Result<ParserError, LiveMatch> => {
+	if (!feMatchDay.liveMatch) {
+		return error({
+			kind: 'ExpectedMatchDayLive',
+			message: `A live match day must have 'liveMatch' set to 'true', but this was not the case for ${feMatchDay.id}`,
+		});
+	}
+
+	const date = parseMatchDate(feMatchDay.date);
+
+	if (date.kind === 'error') {
+		return error({ kind: 'FootballMatchInvalidDate', message: date.error });
+	}
+
+	const homeScore = parseScore(feMatchDay.homeTeam, 'Live');
+
+	if (homeScore.kind === 'error') {
+		return homeScore;
+	}
+
+	const awayScore = parseScore(feMatchDay.awayTeam, 'Live');
+
+	if (awayScore.kind === 'error') {
+		return awayScore;
+	}
+
+	return ok({
+		kind: 'Live',
+		homeTeam: {
+			name: feMatchDay.homeTeam.name,
+			score: homeScore.value,
+		},
+		awayTeam: {
+			name: feMatchDay.awayTeam.name,
+			score: awayScore.value,
+		},
+		dateTime: date.value,
+		paId: feMatchDay.id,
+		comment: feMatchDay.comments,
+		status: feMatchDay.matchStatus,
+	});
+};
+
+const parseMatchDay = (
+	feMatchDay: FEMatchDay,
+): Result<ParserError, FootballMatch> => {
+	const result = oneOf<ParserError, FEMatchDay, FootballMatch>([
+		parseLiveMatch,
+		parseMatchResult,
+		parseFixture,
+	])(feMatchDay);
+
+	if (result.kind === 'error') {
+		return error({ kind: 'InvalidMatchDay', errors: result.error });
+	}
+
+	return result;
+};
+
+const parseMatch = (
+	feMatch: FEFootballMatch,
+): Result<ParserError, FootballMatch> => {
+	switch (feMatch.type) {
+		case 'Fixture':
+			return parseFixture(feMatch);
+		case 'Result':
+			return parseMatchResult(feMatch);
+		case 'MatchDay':
+			return parseMatchDay(feMatch);
+		case 'LiveMatch':
```

Comment:
What is 'LiveMatch'?

Reply:
In frontend we have 4 types for the Football matches: live match, fixture, result & match day. By testing the real data, we realised the live match isn't returned from front end. And if a game is live, frontend always returns MatchDay type but it also has isLive property set to true. So here, it's putting the case for LiveMatch, because this type exists, but it should never be  in the response.